src/scripts/main.js

```diff
@@ -1 +1,17 @@
 'use strict';
+
+const gallery = document.querySelector('.gallery');
+const largeImg = document.getElementById('largeImg');
+
+function mainImage(e) {
+  e.preventDefault();
+
+  const thumb = e.target.closest('.gallery__thumb');
```

Comment:
Ensure that `e.target.closest('.gallery__thumb')` returns a valid element. If the click event does not originate from a `.gallery__thumb`, `thumb` will be `null`, leading to an error on the next line when trying to access `thumb.parentElement`. Consider adding a check to ensure `thumb` is not `null` before proceeding.